### PR TITLE
RPC: Chainparams: Remove Chainparams::fTestnetToBeDeprecatedFieldRPC

### PR DESCRIPTION
**Pull request by jtimon**
Deprecate the "testnet" field in some info RPC calls in favor of the more generic "chain".

Unfortunately #4333 didn't add a "chain" field for rpcmic::getinfo like it did for rpcmining::getmininginfo.
fTestnetToBeDeprecatedFieldRPC was created in #4802.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
This needs mention in the release notes. Even though this was deprecated internally a while ago, users may not have noticed... What about keeping both (the old "testnet" and the new "chain") for a while and mark the "testnet" one as deprecated in the help texts? Then remove "testnet" in 0.15

**Comment by jtimon** [comment]
@MarcoFalke yes, that (having both) was done for rpcmining::getmininginfo but not for rpcmic::getinfo (see https://github.com/bitcoin/bitcoin/pull/4333#issuecomment-45882887 ). IIRC we mentioned in the release notes at the time that testnet would be eventually removed in favor of chain. Maybe @Diapolo (who coded #4326)  remembers better?
At least for rpcmining::getmininginfo I think we're safe taking it out for 0.14. rpcmic::getinfo hasn't died yet, but since @laanwj said that was the plan, I don't think that should be a blocker.

**Comment by laanwj** [comment]
Please, do not make changes to getinfo(). It's going to be removed at some point (now that it's been deprecated), let's not burden anyone with API changes on it until then. at that point you can freely remove this method.

**Comment by MarcoFalke** [comment]
Agree, but you could edit the help text of getinfo to tell where the
replacement can be found.

**Comment by laanwj** [comment]
Changing the help text of `getinfo` to point to the new field on `getblockchaininfo`, would be great.

We could even include this table: https://github.com/bitcoin/bitcoin/issues/8455#issuecomment-250468383 (which I've programmatically verified)

**Comment by jtimon** [comment]
Updated without touching "getinfo" given that is to be removed for 0.15.
I dislike leaving ```Params().NetworkIDString() == CBaseChainParams::TESTNET``` there but since it is only one place that is going to be removed, I think it's fine.

**Comment by MarcoFalke** [comment]
utACK 02eeffac5f686ae59be3db79761e55c12b29d9c9

File: doc/release-notes.md [hunk 1]
```diff
@@ -65,6 +65,8 @@ contain invalid parameters.
 
 The first boolean argument to `getaddednodeinfo` has been removed. This is an incompatible change.
 
+Call "getmininginfo" loses the "testnet" field in favor of the more generic "chain" (which has been having for years).
```
Comment (laanwj) [inline doc/release-notes.md]: "which has been having" I'd reword that as "which has been present"

**Comment by laanwj** [comment]
utACK - language nit

**Comment by jtimon** [comment]
Updated with release notes corrected.
